concrete-core-fixture/src/generation/synthesizing/glwe_ciphertext.rs

```diff
@@ -63,4 +101,223 @@ mod backend_core {
+        }
+
+        fn destroy_glwe_ciphertext_view(
+            &mut self,
+            entity: GlweCiphertextView64,
+            drop_underlying_memory: bool,
+        ) {
+            // We don't know if this the last view referring to the slice it contains, so let the
+            // programmer tell us if that's the case.
+            if drop_underlying_memory {
+                // If requested re-construct the vector so that it frees memory when it's dropped
+                let slice = self
+                    .core_engine
+                    .consume_retrieve_glwe_ciphertext(entity)
+                    .unwrap();
+                unsafe {
+                    Vec::from_raw_parts(slice.as_ptr() as *mut u64, slice.len(), slice.len())
+                };
+            } else {
+                self.core_engine.destroy(entity).unwrap();
+            }
+        }
+    }
+
+    use crate::generation::synthesizing::SynthesizesGlweCiphertextMutView;
+    use concrete_core::prelude::{GlweCiphertextMutView32, GlweCiphertextMutView64};
+
+    impl<'a> SynthesizesGlweCiphertextMutView<Precision32, GlweCiphertextMutView32<'a>> for Maker {
+        fn synthesize_glwe_ciphertext_mut_view(
+            &mut self,
+            prototype: &Self::GlweCiphertextProto,
+        ) -> GlweCiphertextMutView32<'a> {
+            let ciphertext = prototype.0.to_owned();
+            let polynomial_size = ciphertext.polynomial_size();
+            let container = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(ciphertext)
+                .unwrap();
+            self.core_engine
+                .create_glwe_ciphertext(container.leak(), polynomial_size)
+                .unwrap()
+        }
+
+        fn unsynthesize_glwe_ciphertext_mut_view(
+            &mut self,
+            entity: GlweCiphertextMutView32,
+        ) -> Self::GlweCiphertextProto {
+            let polynomial_size = entity.polynomial_size();
+            let slice = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(entity)
+                .unwrap();
+            let ciphertext = self
+                .core_engine
+                .create_glwe_ciphertext(
+                    unsafe { Vec::from_raw_parts(slice.as_mut_ptr(), slice.len(), slice.len()) },
+                    polynomial_size,
+                )
+                .unwrap();
+            ProtoBinaryGlweCiphertext32(ciphertext)
+        }
+
+        fn destroy_glwe_ciphertext_mut_view(&mut self, entity: GlweCiphertextMutView32) {
+            // There should be only one mutable reference to memory, so re-construct the vector so
+            // that it frees memory when it's dropped
+            let slice = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(entity)
+                .unwrap();
+            unsafe { Vec::from_raw_parts(slice.as_mut_ptr(), slice.len(), slice.len()) };
```

Comment:
Same question here: why is it not destroy that's called?

Reply:
Same thing about memory as above but here the slice is immutable so following rust rules we know there should be only one (more means people are doing very unsafe stuff) so we can recreate the vec for the underlying memory to be freed properly

Reply:
this won't be a problem with the reworked design thanks to Alex, resolving